game/cards/dm06/rock_beast.go

```diff
@@ -0,0 +1,51 @@
+package dm06
+
+import (
+	"duel-masters/game/civ"
+	"duel-masters/game/cnd"
+	"duel-masters/game/family"
+	"duel-masters/game/fx"
+	"duel-masters/game/match"
+)
+
+func RumblesaurQ(c *match.Card) {
+
+	c.Name = "Rumblesaur Q"
+	c.Power = 3000
+	c.Civ = civ.Fire
+	c.Family = []string{family.RockBeast, family.Survivor}
+	c.ManaCost = 6
+	c.ManaRequirement = []string{civ.Fire}
+
+	c.Use(fx.Creature, fx.SpeedAttacker, fx.Survivor, fx.When(fx.Summoned, func(card *match.Card, ctx *match.Context) {
+
+		ctx.Match.ApplyPersistentEffect(func(ctx2 *match.Context, exit func()) {
```

Comment:
We can avoid using a persistent effect here since we don't need to do anything in particular when the card leaves the battlezone. Persistent effects are much more computationally expensive than just listening to an event so we should try to only use it where necessary.

Instead we can just listen for the same event as `fx.SpeedAttacker` and remove the summoning sickness of other survivors when they enter the battlefield, and also remove from cards already in the battlezone when this card is summoned.